software/webapp/src/main/java/org/apache/brooklyn/entity/webapp/JavaWebAppSshDriver.java

```diff
@@ -93,11 +93,11 @@ protected String inferRootUrl() {
         if (isProtocolEnabled("https")) {
             Integer port = getHttpsPort();
             checkNotNull(port, "HTTPS_PORT sensors not set; is an acceptable port available?");
-            return String.format("https://%s:%s/", getSubnetHostname(), port);
+            return String.format("https://%s:%s/", getHostname(), port);
```

Comment:
This was already changed from `getHostname` to `getSubnetHostname` to work in clocker env, so probably not safe to just change it back. Given that the value of "main.uri" (the value set here) is the URL we see in the summary tab @bostko's change seems like what the user would expect. So
  * +1 for the change, but better use `BrooklynAccessUtils.getBrooklynAccessibleAddress` to be even more location independent
  * we should figure out how the change fails in the clocker case (would be nice to add a test case) and figure out an alternative solution
  * previously changed in https://github.com/apache/incubator-brooklyn/pull/554, some interesting comments there

@bostko Could you give some more details as to what is the underlying problem leading to this change.